### PR TITLE
Delete synchro conditionals, to clean rows if the configuration is changed

### LINKED ISSUE #9132: Syscollector previous results are not deleted after disabling the feature.
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.2.0-RC9 | syscollector | Manager/Agent | Packages/Sources | All |

## Description

Actual result: It always shows the scan time of the first scan.
Expected result: do not show data after a scan and that the feature is disabled.

Reproduce steps:
- Install Wazuh agent RC9
- Wait to the first one scan.
- Disable process or packages from agent configuration
- Restart agent
- Wait to the first one scan after agent restart
- Check kibana ui

## DoD
- [x] Force different checksum for each scan.
- [x] Manual test
- [x] PR CI Pass

### PR DESCRIPTION
**Pull request by Dwordcito**
|Related issue|
|---|
|Closes #9132 |

## Description
This issue happens since the last scan is not being deleted, after it was disabled and a scan was made again.

Actual result: It always shows the scan time of the first scan.
Expected result: do not show data after a scan and that the feature is disabled.

Reproduce steps:
- Install Wazuh agent RC9
- Wait to the first one scan.
- Disable process or packages from agent configuration
- Restart agent
- Wait to the first one scan after agent restart
- Check 

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [ ] MAC OS X
- [x] Source installation
- [x] Package installation
- [ ] Source upgrade
- [ ] Package upgrade
- [ ] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [ ] Scan-build report
  - [ ] Coverity
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [ ] Dr. Memory
  - [x] AddressSanitizer
- Memory tests for Windows
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Dr. Memory
- Memory tests for macOS
  - [ ] Scan-build report
  - [ ] Leaks
  - [ ] AddressSanitizer

- [ ] Retrocompatibility with older Wazuh versions
- [ ] Working on cluster environments
- [ ] Configuration on demand reports new parameters
- [ ] The data flow works as expected (agent-manager-api-app)
- [ ] Added unit tests (for new features)
- [ ] Stress test for affected components

### CODE REVIEW

**Review by konangc87 — APPROVED** [review]

**Review by jnasselle — APPROVED** [review]

**Comment by tinachoB** [comment]
## Validation

- I've Installed 4.2 agent and manager versions (from the current branch: `dev-resync-with-empty-tables`).
- Agent has been successfully registered and started.
- All information has been synchronized with the manager:
image
- `packages` information has been disabled from the agent side. 
image
- The agent has been restarted.
- All information has been synchronized with the manager (considering the packages disabled option):
image

So, to sum up, **the changes work as expected.**
